src/components/ListingCard.js

```diff
@@ -1,24 +1,52 @@
-import React from "react";
+import React, { useState } from "react"
+
+function ListingCard({
+  listing: { id, description, image, location },
+  onDelete,
+}) {
+  const [favorite, setFavorite] = useState(false)
+
+  function handleFavoriteClick() {
+    setFavorite(!favorite)
+  }
+
+  function handleDeleteClick() {
+    fetch(`http://localhost:6001/listings/${id}`, {
+      method: "DELETE",
+    }).then((response) => response.json())
+    onDelete(id)
+  }
 
-function ListingCard() {
   return (
     <li className="card">
       <div className="image">
         <span className="price">$0</span>
-        <img src={"https://via.placeholder.com/300x300"} alt={"description"} />
+        <img src={image} alt={description} />
       </div>
       <div className="details">
-        {true ? (
-          <button className="emoji-button favorite active">★</button>
+        {favorite ? (
+          <button
+            onClick={handleFavoriteClick}
+            className="emoji-button favorite active"
+          >
+            ★
+          </button>
         ) : (
-          <button className="emoji-button favorite">☆</button>
+          <button
+            onClick={handleFavoriteClick}
+            className="emoji-button favorite"
+          >
+            ☆
+          </button>
```

Comment:
Instead of wrapping this whole thing in a conditional, you could have a conditional for JUST the star. It could be inline in the render or the logic could be in a function outside of the render. Make sense?